staging/src/k8s.io/apiserver/pkg/storage/cacher/watch_cache.go

```diff
@@ -501,29 +532,7 @@ func (s sortableStoreElements) Swap(i, j int) {
 
 // WaitUntilFreshAndList returns list of pointers to `storeElement` objects along
 // with their ResourceVersion and the name of the index, if any, that was used.
-func (w *watchCache) WaitUntilFreshAndList(ctx context.Context, resourceVersion uint64, key string, matchValues []storage.MatchValue) ([]interface{}, uint64, string, error) {
```

Comment:
Can we leave all of this in place and instead gate changes in here or duplicate a separate method for the new approach?

Reply:
Most of the code change here is refactor, the gated code is just:
```
if store, ok := w.store.(orderedLister); ok {
	result, _ := store.ListPrefix(key, "", 0)
	return result, w.resourceVersion, "", nil
}
```

Would a separate commit or PR with refactor be ok?

Reply:
Extracted https://github.com/kubernetes/kubernetes/pull/126968